ios/FluentUI.Demo/FluentUI.Demo/Demos/ColorDemoController.swift

```diff
@@ -146,30 +141,28 @@ class ColorDemoController: UIViewController {
 
     @objc private func segmentedControlValueChanged(sender: Any) {
         if let segmentedControl = sender as? SegmentedControl {
-            let windowType = colorProviderThemes[segmentedControl.selectedSegmentIndex].demoColorTheme.windowType
-            let colorThemeHost = view.window?.windowScene?.delegate as? ColorThemeHosting
-            ColorDemoController.themeWindowType = windowType
-
-            if let navigationController = navigationController {
-                navigationController.popViewController(animated: false)
-                colorThemeHost?.updateToWindowWith(type: windowType, pushing: self)
+            let selectedSegmentIndex = segmentedControl.selectedSegmentIndex
+            let provider = colorProviderThemes[selectedSegmentIndex].demoColorTheme.provider
+            let window = self.view.window
+            ColorDemoController.providerType = colorProviderThemes[selectedSegmentIndex].demoColorTheme.providerType
+
+            if let provider = provider as? ColorProviding {
```

Comment:
this can be cleaned up as let provider = provider ?? DemoColorDefaultTheme()  so we reduce the duplicated code? or lets just get rid of none case if you are going to do it this way? the intent of the "none" segment control is that the client doesn't give any provider so the colors are all fall back. in your previous iteration, you actually had a remove provider of the dictionary?